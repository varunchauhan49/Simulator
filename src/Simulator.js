import React, { Component } from 'react';
import Board from './Board';
import UserInput from './UserInput';
import {Row,Col,Jumbotron} from 'reactstrap';

/*
Simulator class is main entry point for Robot Simulation module.
*/
class Simulator extends Component{
  constructor(props){
    super();
    this.state = {
      stepX:-1,
      stepY:-1,
      stepFace:''
    };
  }
  handleAction(x,y,face){
    this.setState({
      stepX:x,
      stepY:y,
      stepFace:face
    })
  }
  render(){
    let {stepX,stepY,stepFace} = this.state;
    return(
      <Row>
        <Col xs="5">
          <Row style={{padding:'0px 0px 20px 20px',marginBottom:'5vh'}}>
            <Jumbotron>
              <h3>Robot Simulator</h3>
              <p>Orientation : {stepFace}</p>
              <p>X: {stepX}</p>
              <p>Y: {stepY}</p>
              <hr className="my-2" />
              <p>A robot simulator which allows you to move robot across the board.</p>
            </Jumbotron>
          </Row>
          <Row>
            <UserInput 
              action={(x,y,face) => this.handleAction(x,y,face)}
              rows={8}
              cols={8} 
              />
          </Row>
        </Col>
        <Col xs="7">
          <Board 
            posX={stepX}
            posY={stepY}
            face={stepFace}
            rows={8}
            cols={8}
            />
        </Col>
      </Row>
    );
  }
}

export default Simulator;