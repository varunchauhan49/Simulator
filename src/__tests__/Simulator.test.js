import { shallow } from 'enzyme';
import Simulator from '../Simulator';
import React from 'react';

describe('Simulator', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Simulator />);
    expect(wrapper).toMatchSnapshot();
  });
  it('evaluate state of class simulator when data passed bu userInput', () => {
    const wrapper = shallow(<Simulator />);
    const instance = wrapper.instance();
    instance.handleAction(1,1,'north');
    expect(wrapper.state().stepX).toBe(1);
    expect(wrapper.state().stepY).toBe(1);
    expect(wrapper.state().stepFace).toBe('north');
  });
});