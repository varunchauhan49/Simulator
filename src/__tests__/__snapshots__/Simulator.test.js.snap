// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simulator renders correctly 1`] = `
<Row
  tag="div"
>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="5"
  >
    <Row
      style={
        Object {
          "marginBottom": "5vh",
          "padding": "0px 0px 20px 20px",
        }
      }
      tag="div"
    >
      <Jumbotron
        tag="div"
      >
        <h3>
          Robot Simulator
        </h3>
        <p>
          Orientation : 
        </p>
        <p>
          X: 
          -1
        </p>
        <p>
          Y: 
          -1
        </p>
        <hr
          className="my-2"
        />
        <p>
          A robot simulator which allows you to move robot across the board.
        </p>
      </Jumbotron>
    </Row>
    <Row
      tag="div"
    >
      <UserInput
        action={[Function]}
        cols={8}
        rows={8}
      />
    </Row>
  </Col>
  <Col
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="7"
  >
    <Board
      cols={8}
      face=""
      posX={-1}
      posY={-1}
      rows={8}
    />
  </Col>
</Row>
`;
